### ISSUE TITLE
ROS2 (Galactic) on Ubuntu 20.04 - no messages published under topic /hesai/pandar and PandarView run error

### ISSUE THREAD

**Issue by danijel-fmci** [HesaiTechnology/HesaiLidar_PandarGeneral_ros#70]
Hi,

I managed to set-up ROS2 build on Ubuntu 20.04 with help of [https://github.com/HesaiTechnology/HesaiLidar_General_ROS/issues/53#issuecomment-1054363780](url)

I have Pandar64 and I can ping it and I can see packets in the Wireshark but no packets published under `/hesai/pandar`

Also, I was not able to run PandarView.sh, please see attached error log.

Could you help me with this issue, please?
Best,
Danijel

![PandarView2](https://user-images.githubusercontent.com/111062102/232080696-16f026df-0f3c-441f-bd31-fbd56896eb5b.png)
![hesai_lidar_launch](https://user-images.githubusercontent.com/111062102/232083308-2d2faef9-1b6d-4a1b-843d-63fceac3aecd.png)
![ro2 topic](https://user-images.githubusercontent.com/111062102/232083327-47fd302c-3e4b-4a12-b490-11e788e2cc43.png)

**Comment by shashankyld** [comment]
Did you install SDK manually prior to building ROS workspace?

**Comment by bharadwajakshay** [comment]
Any updates?? I'm running into the same error with ROS neotic.

**Comment by danijel-fmci** [comment]
> Did you install SDK manually prior to building ROS workspace?

No, I followed the guide on https://github.com/HesaiTechnology/HesaiLidar_General_ROS/tree/ROS2 and then applied solutions from https://github.com/HesaiTechnology/HesaiLidar_General_ROS/issues/53

In the meantime, I switched back to ROS1 as there are fewer issues. Seems like drivers are originally written for ROS1 and then converted to ROS2 after.

**Comment by yangfanFred** [comment]
Hi Danijel,

General_ROS project and Swift_ROS project are officially designed to support ROS1.
Now we have ROS_2.0 project to support ROS2, pelase try this project if you are using ROS2 versions.
The pandarview.sh running error seems caused by graphic card driver compatibllity issue, the easiest way to solve this is to try run pandarview on another PC with a different graphic card.

Thanks,
Fred

**Comment by danijel-fmci** [comment]
[like]  Danijel Benjak reacted to your message:
________________________________
